File: app/src/main/java/com/sumian/sd/widget/pay/PayItemView.java
package com.sumian.sd.widget.pay;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.drawable.Drawable;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.Gravity;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.sumian.sd.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by sm
 * on 2018/1/22.
 * desc:
 */

public class PayItemView extends LinearLayout implements View.OnClickListener {

    @BindView(R.id.iv_pay_icon)
    ImageView mIvPayIcon;

    @BindView(R.id.tv_pay_desc)
    TextView mTvPayDesc;

    @BindView(R.id.iv_pay_select)
    ImageView mIvPay;

    private OnPayWayCallback mOnPayWayCallback;

    public PayItemView(Context context) {
        this(context, null);
    }

    public PayItemView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public PayItemView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
        init(context, attrs, defStyleAttr);
    }

    private void init(Context context, AttributeSet attrs, int defStyleAttr) {
        TypedArray a = context.obtainStyledAttributes(attrs, R.styleable.PayItemView, defStyleAttr, 0);

        Drawable iconDrawable = a.getDrawable(R.styleable.PayItemView_icon);
        this.mIvPayIcon.setImageDrawable(iconDrawable);

        String text = a.getString(R.styleable.PayItemView_desc);
        mTvPayDesc.setText(text);

        boolean isSelect = a.getBoolean(R.styleable.PayItemView_is_select, false);
        mIvPay.setTag(isSelect ? true : null);
        mIvPay.setImageResource(isSelect ? R.mipmap.ic_group_pay_selected : R.mipmap.ic_group_pay_unselected);

        a.recycle();

        setOnClickListener(this);
    }

    private void init(Context context) {
        ButterKnife.bind(inflate(context, R.layout.lay_pay_divider_item, this));
        int padding = (int) context.getResources().getDimension(R.dimen.space_20);
        setPadding(padding, padding, padding, padding);
        setBackgroundColor(getResources().getColor(R.color.b2_color));
        setGravity(Gravity.CENTER);
        setOrientation(HORIZONTAL);
    }

    public void setOnPayWayCallback(OnPayWayCallback onPayWayCallback) {
        mOnPayWayCallback = onPayWayCallback;
    }

    public void select() {
        mIvPay.setTag(true);
        mIvPay.setImageResource(R.mipmap.ic_group_pay_selected);
        if (mOnPayWayCallback != null) {
            mOnPayWayCallback.onSelectPayWay(this);
        }
    }

    public void unSelect() {
        mIvPay.setTag(null);
        mIvPay.setImageResource(R.mipmap.ic_group_pay_unselected);
    }

    // @OnClick({R.id.iv_pay_select})
    @Override
    public void onClick(View v) {
        // switch (v.getId()) {
        //    case R.id.iv_pay_select:
        if (mIvPay.getTag() == null) {
            select();
        }
        //       break;
        //  default:
        //    break;
        // }
    }


    public interface OnPayWayCallback {

        void onSelectPayWay(View v);

    }
}

File: app/src/main/java/com/sumian/sd/pay/widget/PayItemView.kt
@file:Suppress("MemberVisibilityCanBePrivate", "DEPRECATION")

package com.sumian.sd.pay.widget

import android.content.Context
import android.util.AttributeSet
import android.view.Gravity
import android.view.View
import android.widget.LinearLayout
import com.sumian.sd.R
import kotlinx.android.synthetic.main.lay_pay_divider_item.view.*


/**
 * Created by sm
 * on 2018/1/22.
 * desc:
 */

class PayItemView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : LinearLayout(context, attrs, defStyleAttr), View.OnClickListener {

    private var mOnPayWayCallback: OnPayWayCallback? = null

    init {
        init(context)
        init(context, attrs, defStyleAttr)
    }

    private fun init(context: Context, attrs: AttributeSet?, defStyleAttr: Int) {
        val a = context.obtainStyledAttributes(attrs, R.styleable.PayItemView, defStyleAttr, 0)

        val iconDrawable = a.getDrawable(R.styleable.PayItemView_icon)
        iv_pay_icon.setImageDrawable(iconDrawable)

        val text = a.getString(R.styleable.PayItemView_desc)
        tv_pay_desc.text = text

        val isSelect = a.getBoolean(R.styleable.PayItemView_is_select, false)
        iv_pay_select.tag = if (isSelect) true else null
        iv_pay_select.setImageResource(if (isSelect) R.mipmap.ic_group_pay_selected else R.mipmap.ic_group_pay_unselected)

        a.recycle()

        setOnClickListener(this)
    }

    private fun init(context: Context) {
        gravity = Gravity.CENTER
        orientation = LinearLayout.HORIZONTAL
        View.inflate(context, R.layout.lay_pay_divider_item, this)
        val padding = context.resources.getDimension(R.dimen.space_20).toInt()
        setPadding(padding, 0, padding, 0)
        setBackgroundColor(resources.getColor(R.color.b2_color))
    }

    fun setOnPayWayCallback(onPayWayCallback: OnPayWayCallback) {
        mOnPayWayCallback = onPayWayCallback
    }

    fun select() {
        iv_pay_select.tag = true
        iv_pay_select.setImageResource(R.mipmap.ic_group_pay_selected)
        mOnPayWayCallback?.onSelectPayWay(this)
    }

    fun unSelect() {
        iv_pay_select.tag = null
        iv_pay_select.setImageResource(R.mipmap.ic_group_pay_unselected)
    }

    override fun onClick(v: View) {
        if (iv_pay_select.tag == null) {
            select()
        }
    }


    interface OnPayWayCallback {

        fun onSelectPayWay(v: View)

    }
}
